kimchi/src/circuits/lookup/constraints.rs

```diff
@@ -219,29 +217,29 @@ where
 /// after multiplying all of the values, all of the terms will have cancelled if s is a sorting of f and t, and the final term will be 1
 /// because of the random choice of beta and gamma, there is negligible probability that the terms will cancel if s is not a sorting of f and t
 #[allow(clippy::too_many_arguments)]
-pub fn aggregation<R, F>(
-    dummy_lookup_value: F,
-    joint_lookup_table_d8: &Evaluations<F, D<F>>,
-    d1: D<F>,
-    gates: &[CircuitGate<F>],
-    witness: &[Vec<F>; COLUMNS],
-    joint_combiner: &F,
-    table_id_combiner: &F,
-    beta: F,
-    gamma: F,
-    sorted: &[Evaluations<F, D<F>>],
+pub fn aggregation<R, G>(
+    dummy_lookup_value: G::ScalarField,
+    joint_lookup_table_d8: &Evaluations<G::ScalarField, D<G::ScalarField>>,
+    d1: D<G::ScalarField>,
+    gates: &[CircuitGate<G>],
+    witness: &[Vec<G::ScalarField>; COLUMNS],
+    joint_combiner: &G::ScalarField,
+    table_id_combiner: &G::ScalarField,
+    beta: G::ScalarField,
+    gamma: G::ScalarField,
+    sorted: &[Evaluations<G::ScalarField, D<G::ScalarField>>],
     rng: &mut R,
     lookup_info: &LookupInfo,
-) -> Result<Evaluations<F, D<F>>, ProverError>
+) -> Result<Evaluations<G::ScalarField, D<G::ScalarField>>, ProverError>
 where
     R: Rng + ?Sized,
```

Comment:
unrelated to this PR, but looking at this I think we should do what we did elsewhere in the codebase (`rng: &mut (impl RngCore + CryptoRng)`)

Reply:
I think that it could be a good candidate for `easy task to start contributing`.

Reply:
https://github.com/o1-labs/proof-systems/issues/673